packages/svelte/src/reactivity/map.js

```diff
@@ -126,53 +125,23 @@ export class ReactiveMap extends Map {
 
 	keys() {
 		get(this.#version);
-		var iterator = this.#sources.keys();
-
-		return make_iterable(
-			/** @type {IterableIterator<K>} */ ({
-				next() {
-					for (var value of iterator) {
-						return { value, done: false };
-					}
-
-					return { done: true };
-				}
-			})
-		);
+		return this.#sources.keys();
 	}
 
-	values() {
+	*values() {
 		get(this.#version);
-		var iterator = this.#sources.values();
-
-		return make_iterable(
-			/** @type {IterableIterator<V>} */ ({
-				next() {
-					for (var source of iterator) {
-						return { value: get(source), done: false };
-					}
-
-					return { done: true };
-				}
-			})
-		);
+
+		for (var source of this.#sources.values()) {
+			yield get(source);
```

Comment:
Generator functions are notoriously terrible for performance.

Reply:
I doubt it makes much difference in real situations, but I updated the PR with a `map` utility instead